corehq/project_limits/rate_limiter.py

```diff
@@ -50,15 +50,26 @@ def allow_usage(self, scope=''):
         allowed = False
         # allow usage if any scope has capacity
         for limit_scope, rates in self.iter_rates(scope):
-            allow = all(current_rate < limit
-                        for rate_counter_key, current_rate, limit in rates)
+            allow = all(rate_counter.get(self.feature_key + limit_scope) < limit
+                        for rate_counter, limit in rates)
             # for each scope all counters must be below threshold
             if allow:
                 allowed = True
             else:
                 metrics_counter('commcare.rate_limit_exceeded', tags={'key': self.feature_key, 'scope': scope})
         return allowed
 
+    def get_retry_after(self, scope):
+        retry_after = 0.0
+        for limit_scope, rates in self.iter_rates(scope):
+            if limit_scope != scope:
```

Comment:
Apologies for the radio silence. I updated this part of the code, so this check isn't necessary. See https://github.com/dimagi/commcare-hq/pull/33572/commits/438131d608b42cdc93a9d58656ec1bd79714dac5.

But to answer your question: In [this part](https://github.com/dimagi/commcare-hq/blob/438131d608b42cdc93a9d58656ec1bd79714dac5/corehq/project_limits/rate_limiter.py#L159) where we call `get_n_users_in_subscription`, it returns `Dimagi Internal Test Account for Project {your-domain-name}` for domains that's on the Internal Test Subscription. In that function we call
```python
return plan_version.feature_rates.get(feature__feature_type='User').monthly_limit
```
where the `monthly_limit` for this plan is "-1". This in turn makes [this code](https://github.com/dimagi/commcare-hq/blob/438131d608b42cdc93a9d58656ec1bd79714dac5/corehq/project_limits/rate_limiter.py#L168-L172) return a negative threshold / limit. This isn't an issue for [the method that checks if the user is allowed](https://github.com/dimagi/commcare-hq/blob/438131d608b42cdc93a9d58656ec1bd79714dac5/corehq/project_limits/rate_limiter.py#L49-L60) to make calls, but it was an issue for my method, since that will always require a very high `retry_time`, causing my `max` function to always return its high value for the retry time.

My update in https://github.com/dimagi/commcare-hq/pull/33572/commits/438131d608b42cdc93a9d58656ec1bd79714dac5 basically keeps track of the retry times required across all scopes and selects the minimum retry time. We can do this, since 1. the `allow_usage` method allows you to proceed with the call if _any_ scope has capacity and 2. the minimum time to wait across all scopes will allow at least 1 scope to get capacity.